src/Cli/dotnet/commands/dotnet-sln/remove/Program.cs

```diff
@@ -18,16 +18,27 @@ internal class RemoveProjectFromSolutionCommand : CommandBase
         private readonly string _fileOrDirectory;
         private readonly IReadOnlyCollection<string> _arguments;
 
-        public RemoveProjectFromSolutionCommand(
-            ParseResult parseResult) : base(parseResult)
+        public RemoveProjectFromSolutionCommand(ParseResult parseResult) : base(parseResult)
         {
-            _arguments = (parseResult.ValueForArgument<IEnumerable<string>>(SlnRemoveParser.ProjectPathArgument) ?? Array.Empty<string>()).ToList().AsReadOnly();
+            _fileOrDirectory = parseResult.ValueForArgument<string>(SlnCommandParser.SlnArgument);
+
+            _arguments = parseResult.ValueForArgument(SlnRemoveParser.ProjectPathArgument)?.ToArray() ?? (IReadOnlyCollection<string>)Array.Empty<string>();
             if (_arguments.Count == 0)
             {
                 throw new GracefulException(CommonLocalizableStrings.SpecifyAtLeastOneProjectToRemove);
             }
 
-            _fileOrDirectory = parseResult.ValueForArgument<string>(SlnCommandParser.SlnArgument);
+            var slnFile = _arguments.FirstOrDefault(path => path.EndsWith(".sln"));
+            if (slnFile != null)
```

Comment:
Same as above, we can factor this section into a method. In fact, we can probably reuse the same method across both commands.